Introduce a generic Exit/Error handler and use this for the login command

lloyddave opened the pull request:
The error_handler package provides function `ErrorExit`; the function prints
out an error message to `STDERR` and exits with a given or default exit code.
The first argument can be of type `string` or `error`.  The second argument
(optional) should be of type `integer` and is for setting the exit code.
Three integer constants are defined for setting the exit code:
- `CLIUsageErrorExitCode` - for reporting command syntax errors
- `CliGenericErrorExitCode` - for reporting processing errors
- `CLITrapErrorCode` - for reporting abnormal/trapped errors
  If the second argument is not provided the `CliGenericErrorExitCode` value is used.

The `login` command has been extended to connect to Brooklyn and obtain the version string which is then printed to `STDOUT`.

robertgmoss commented on api/login/login.go:
```diff
@@ -0,0 +1,15 @@
+	"github.com/brooklyncentral/brooklyn-cli/net"
+)
+
+func Login(network *net.Network) string {
```
There is already a server api package with this endpoint

lloyddave replied:
I wasn't quite sure what would end up in the package at the start.  But since it is now effectively a duplicate of version I will remove this to avoid the duplication.

robertgmoss commented on commands/login.go:
```diff
@@ -66,4 +61,7 @@ func (cmd *Login) Run(c *cli.Context) {
 	cmd.config.Map["target"] = cmd.network.BrooklynUrl
 	cmd.config.Write()
+	
+	loginversion := login.Login(cmd.network)
```
use the server api command instead

robertgmoss commented:
Looks good, only one minor comment.  Needs to be rebased now too.

lloyddave committed:
Introduce a generic Exit/Error handler and use this for the login command

The error_handler package provides function ErrorExit; the function prints
out an error message to STDERR and exits with a given or default exit code.
The first argument can be of type string or error.  The second argument
(optional) should be of type integer and is for setting the exit code.
Three integer constants are defined for setting the exit code:
  CLIUsageErrorExitCode - for reporting command syntax errors
  CliGenericErrorExitCode - for reporting processing errors
  CLITrapErrorCode - for reporting abnormal/trapped errors
If the second argument is not provided the CliGenericErrorExitCode value is used.

The login command has been extended to connect to Brooklyn and obtain the version
string which is then printed to STDOUT.

lloyddave committed:
Removed import for "os"

geomacy commented on commands/login.go:
```diff
@@ -67,4 +62,7 @@ func (cmd *Login) Run(scope scope.Scope, c *cli.Context) {
 
 	cmd.config.Map["target"] = cmd.network.BrooklynUrl
 	cmd.config.Write()
+	
```
Maybe do the login before writing the config and only do the latter if the former succeeds.

lloyddave replied:
I tried only writing the config if the login succeeds, but this has a bad side affect; if the login fails, the previous credentials are still enabled in the config file and potentially pointing to the wrong brooklyn node.  Need more thought on what would be the best approach.

geomacy commented:
Let's accept this change as a work in progress and deal with the issue [above](https://github.com/brooklyncentral/brooklyn-cli/pull/6#discussion_r47072246) as a separate matter.
